HLTrigger/HLTcore/interface/TriggerExpressionParser.h

```diff
@@ -2,7 +2,7 @@
 #define HLTrigger_HLTfilters_TriggerExpressionParser_h
 
 // Note: this requires Boost 1.41 or higher, for Spirit 2.1 or higher
-#include <boost/spirit/include/phoenix.hpp>
+#include <boost/phoenix.hpp>
```

Comment:
This change quiets the following compile-time warning:
```
In file included from /cvmfs/cms.cern.ch/slc7_amd64_gcc10/external/boost/1.78.0-12075919175e8d078539685f9234134a/include/boost/config/header_deprecated.hpp:18,
                 from /cvmfs/cms.cern.ch/slc7_amd64_gcc10/external/boost/1.78.0-12075919175e8d078539685f9234134a/include/boost/spirit/include/phoenix.hpp:11,
                 from CMSSW_12_5_0_pre4/src/HLTrigger/HLTcore/interface/TriggerExpressionParser.h:5,
                 from CMSSW_12_5_0_pre4/src/HLTrigger/HLTcore/test/test_catch2_TriggerExpressionParser.cc:7:
/cvmfs/cms.cern.ch/slc7_amd64_gcc10/external/boost/1.78.0-12075919175e8d078539685f9234134a/include/boost/spirit/include/phoenix.hpp:12:1: note: '#pragma message: This header is deprecated. Use <boost/phoenix.hpp> instead.'
   12 | BOOST_HEADER_DEPRECATED("<boost/phoenix.hpp>")
      | ^~~~~~~~~~~~~~~~~~~~~~~
```
It seemed safe to do, but I'm not 100% sure. @fwyzard , what do you think?

Reply:
If it works and silences the warnings, go for it